tox.ini

```diff
@@ -1,56 +1,89 @@
 [tox]
 envlist =
-    py27,py35,py36,py37,py38,pypy,pypy3,
+    lint,
+    py27,pypy,
+    py35,py36,py37,py38,pypy3,
     docs,
-    {py27,py35}-cover,coverage
+    coverage
+isolated_build = True
 
 [testenv]
 commands =
     python --version
-    nosetests --with-xunit --xunit-file=nosetests-{envname}.xml {posargs:waitress}
-
+    pytest {posargs:}
 extras =
     testing
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
 
-[testenv:docs]
-basepython = python3.5
-whitelist_externals = make
+[testenv:coverage]
+basepython = python3.8
 commands =
-    make -C docs clean html epub BUILDDIR={envdir} "SPHINXOPTS=-W -E"
-extras =
-    docs
+    coverage combine
+    coverage xml
+    coverage report --show-missing
+deps =
+    coverage
+setenv =
+    COVERAGE_FILE=.coverage
+depends = py27, py35
 
-[py-cover]
+[testenv:lint]
+skip_install = True
+basepython = python3.8
 commands =
-    coverage run {envbindir}/nosetests waitress
-    coverage combine
-    coverage xml -o {envname}.xml
+    black --check --diff .
+    check-manifest
+    # build sdist/wheel
+    python -m pep517.build .
+    twine check dist/*
+deps =
+    black
+    readme_renderer
+    check-manifest
+    pep517
+    twine
 
+[testenv:docs]
+whitelist_externals =
+    make
+commands =
+    make -C docs html BUILDDIR={envdir} SPHINXOPTS="-W -E"
 extras =
-    testing
+    docs
 
-[testenv:py27-cover]
+[testenv:run-flake8]
+skip_install = True
+basepython = python3.8
 commands =
-    {[py-cover]commands}
-
-setenv =
-    COVERAGE_FILE=.coverage.py2
+    flake8 src/waitress/ tests
+deps =
+    flake8
+    flake8-bugbear
```

Comment:
flake8 should be part of the lint step - black does not cover everything that flake8 verifies

Reply:
:+1:

Reply:
This PR is still missing a flake8 config too.

Reply:
I remember why I didn't add that in the first place... this repo does not yet pass flake8 :P